wasmtime-wasi/src/host.rs

```diff
@@ -4,3 +4,140 @@ include!(concat!(env!("OUT_DIR"), "/wasmtime_ssp.rs"));
 
 pub type char = ::std::os::raw::c_char;
 pub type void = ::std::os::raw::c_void;
+
+use super::wasm32;
+
+// Taken from rust's implementation of `std::os` functionality for unix systems
+// rust/src/libstd/sys/unix/os.rs
+// https://github.com/rust-lang/rust/blob/9ebf47851a357faa4cd97f4b1dc7835f6376e639/src/libstd/sys/unix/os.rs
+extern "C" {
+    #[cfg(not(target_os = "dragonfly"))]
+    #[cfg_attr(
+        any(target_os = "linux", target_os = "fuchsia", target_os = "l4re"),
+        link_name = "__errno_location"
+    )]
+    #[cfg_attr(
+        any(
+            target_os = "bitrig",
+            target_os = "netbsd",
+            target_os = "openbsd",
+            target_os = "android",
+            target_os = "hermit",
+            target_env = "newlib"
+        ),
+        link_name = "__errno"
+    )]
+    #[cfg_attr(target_os = "solaris", link_name = "___errno")]
+    #[cfg_attr(
+        any(target_os = "macos", target_os = "ios", target_os = "freebsd"),
+        link_name = "__error"
+    )]
+    #[cfg_attr(target_os = "haiku", link_name = "_errnop")]
+    fn errno_location() -> *mut libc::c_int;
+}
```

Comment:
I couldn't decide between using `nix` crate or linking to `errno` ourselves, but in the end, decided to borrow the functionality for *nix directly from Rust's `libstd`.

Reply:
We can use the [errno crate](https://crates.io/crates/errno) for cross-platform access to errno (wasmtime uses this in a few other places already).

Reply:
Ah, nice! I'll investigate today and make amends.

Reply:
OK, we're now using the [errno crate](https://crates.io/crates/errno).